react/features/app/actions.js

```diff
@@ -49,6 +47,15 @@ export function appNavigate(uri) {
                     .domain;
         }
 
+        if (room) {
```

Comment:
This should be part of the set URL commit. Also, can't this be simplified by checking if there is no room and doing it like you were doing it before otherwise? Why is the splitting necessary?